### PR TITLE
Added ``--generate-repo-config`` to to the rpm distribution.

### PR DESCRIPTION
**Pull request by ipanova**
[noissue]

Review Checklist:
- [ ] An issue is properly linked. [feature and bugfix only]
- [ ] Tests are present or not feasible.
- [ ] Commits are split in a logical way (not historically).

### CODE REVIEW

File: pulpcore/cli/rpm/distribution.py [hunk 1]
```diff
@@ -67,6 +67,15 @@ def distribution(ctx: click.Context, pulp_ctx: PulpCLIContext, distribution_type
...
     ),
+    click.option(
+        "--generate-repo-config",
+        type=click.BOOL,
+        default=False,
+        help=_(
+            "Option specifying whether ``*.repo`` files will be generated and served."
+            " Defaults to False."
+        ),
+    ),
```
Comment (mdellweg) [inline pulpcore/cli/rpm/distribution.py:78]: This flag will only appear in pulp_rpm 3.23, right? You should make it a `pulp_option` with `needs_plugins=[PluginRequirement("rpm", specifier=">=3.23.dev")]`.

File: pulpcore/cli/rpm/distribution.py [hunk 2]
```diff
@@ -67,6 +67,15 @@ def distribution(ctx: click.Context, pulp_ctx: PulpCLIContext, distribution_type
             "auto-distribute."
         ),
     ),
+    click.option(
+        "--generate-repo-config",
+        type=click.BOOL,
+        default=False,
```
Comment (mdellweg) [inline pulpcore/cli/rpm/distribution.py:73]:
```suggestion
        "--generate-repo-config/--no-generate-repo-config",
```

File: pulpcore/cli/rpm/distribution.py (hunk 1 above)
Comment (ipanova) [reply pulpcore/cli/rpm/distribution.py:78]: thanks for the tip

File: pulpcore/cli/rpm/distribution.py [hunk 3]
```diff
@@ -67,6 +69,14 @@ def distribution(ctx: click.Context, pulp_ctx: PulpCLIContext, distribution_type
             "auto-distribute."
         ),
     ),
+    pulp_option(
+        "--generate-repo-config/--no-generate-repo-config",
+        needs_plugins=[PluginRequirement("rpm", specifier=">=3.23.0.dev")],
```
Comment (mdellweg) [inline pulpcore/cli/rpm/distribution.py:74]: And now we should remove this line again.

File: pulpcore/cli/rpm/distribution.py [hunk 4]
```diff
@@ -67,6 +68,13 @@ def distribution(ctx: click.Context, pulp_ctx: PulpCLIContext, distribution_type
             "auto-distribute."
         ),
     ),
+    pulp_option(
+        "--generate-repo-config/--no-generate-repo-config",
+        help=_(
+            "Option specifying whether ``*.repo`` files will be generated and served."
+            " Defaults to False."
```
Comment (mdellweg) [inline pulpcore/cli/rpm/distribution.py:75]:
```suggestion
```
It's probably better to conceal this.

File: pulpcore/cli/rpm/distribution.py [hunk 5]
```diff
@@ -67,6 +68,13 @@ def distribution(ctx: click.Context, pulp_ctx: PulpCLIContext, distribution_type
             "auto-distribute."
         ),
     ),
+    pulp_option(
+        "--generate-repo-config/--no-generate-repo-config",
+        help=_(
```
Comment (mdellweg) [inline pulpcore/cli/rpm/distribution.py:73]:
```suggestion
        "--generate-repo-config/--no-generate-repo-config",
        default=None,
        help=_(
```
I guess, this is needed to make the option a tri-state...

**Review by mdellweg — APPROVED** [review]

**Pull request merged**